[SUPPORT] Datadog metric reporter fails with 0.14.1-rc1 as well as 0.14.0

spark 3.2.4 / hudi 0.14.1-rc1
The below code works fine with hudi 0.13.1. Apparently something wrong during the metadata table metrics reporter

```python
tableName = 'test_hudi'
basePath = "/tmp/{tableName}".format(tableName=tableName)
df = (
    spark.sql("select 1 event_id, 2 event_date, 3 version")
)

hudi_options = {
    "hoodie.table.name": tableName,
    "hoodie.datasource.write.recordkey.field": "event_id",
    "hoodie.datasource.write.partitionpath.field": "event_date",
    "hoodie.datasource.write.table.name": tableName,
    "hoodie.datasource.write.operation": "bulk_insert",
    "hoodie.datasource.write.precombine.field": "version",
    "hoodie.combine.before.insert":"true",
    "hoodie.datasource.hive_sync.enable": "false",
    "hoodie.datasource.write.keygenerator.class": "org.apache.hudi.keygen.ComplexKeyGenerator",
    "hoodie.merge.allow.duplicate.on.inserts":"true",
    "hoodie.metadata.enable": "true",
    "hoodie.metrics.on": "true",
    "hoodie.metrics.reporter.type": "DATADOG",
    "hoodie.metrics.datadog.api.key.skip.validation": "true",
    "hoodie.metrics.datadog.metric.prefix": "hudi",
    "hoodie.metrics.datadog.api.site": "US",
    "hoodie.metrics.datadog.api.key": "dummykey",

}
(df.write.format("hudi").options(**hudi_options).mode("overwrite").save(basePath))

```

```
 : org.apache.hudi.exception.HoodieException: Failed to instantiate Metadata table
        at org.apache.hudi.client.SparkRDDWriteClient.initializeMetadataTable(SparkRDDWriteClient.java:293)
        at org.apache.hudi.client.SparkRDDWriteClient.initMetadataTable(SparkRDDWriteClient.java:273)
        at org.apache.hudi.client.BaseHoodieWriteClient.doInitTable(BaseHoodieWriteClient.java:1257)
        at org.apache.hudi.client.BaseHoodieWriteClient.initTable(BaseHoodieWriteClient.java:1297)
        at org.apache.hudi.commit.BaseDatasetBulkInsertCommitActionExecutor.execute(BaseDatasetBulkInsertCommitActionExecutor.java:95)
        at org.apache.hudi.HoodieSparkSqlWriterInternal.bulkInsertAsRow(HoodieSparkSqlWriter.scala:935)
        at org.apache.hudi.HoodieSparkSqlWriterInternal.writeInternal(HoodieSparkSqlWriter.scala:481)
        at org.apache.hudi.HoodieSparkSqlWriterInternal.write(HoodieSparkSqlWriter.scala:204)
        at org.apache.hudi.HoodieSparkSqlWriter$.write(HoodieSparkSqlWriter.scala:121)
        at org.apache.hudi.DefaultSource.createRelation(DefaultSource.scala:150)
        at org.apache.spark.sql.execution.datasources.SaveIntoDataSourceCommand.run(SaveIntoDataSourceCommand.scala:47)
        at org.apache.spark.sql.execution.command.ExecutedCommandExec.sideEffectResult$lzycompute(commands.scala:75)
        at org.apache.spark.sql.execution.command.ExecutedCommandExec.sideEffectResult(commands.scala:73)
        at org.apache.spark.sql.execution.command.ExecutedCommandExec.executeCollect(commands.scala:84)
        at org.apache.spark.sql.execution.QueryExecution$$anonfun$eagerlyExecuteCommands$1.$anonfun$applyOrElse$1(QueryExecution.scala:97)
        at org.apache.spark.sql.execution.SQLExecution$.$anonfun$withNewExecutionId$5(SQLExecution.scala:103)
        at org.apache.spark.sql.execution.SQLExecution$.withSQLConfPropagated(SQLExecution.scala:163)
        at org.apache.spark.sql.execution.SQLExecution$.$anonfun$withNewExecutionId$1(SQLExecution.scala:90)
        at org.apache.spark.sql.SparkSession.withActive(SparkSession.scala:775)
        at org.apache.spark.sql.execution.SQLExecution$.withNewExecutionId(SQLExecution.scala:64)
        at org.apache.spark.sql.execution.QueryExecution$$anonfun$eagerlyExecuteCommands$1.applyOrElse(QueryExecution.scala:97)
        at org.apache.spark.sql.execution.QueryExecution$$anonfun$eagerlyExecuteCommands$1.applyOrElse(QueryExecution.scala:93)
        at org.apache.spark.sql.catalyst.trees.TreeNode.$anonfun$transformDownWithPruning$1(TreeNode.scala:481)
        at org.apache.spark.sql.catalyst.trees.CurrentOrigin$.withOrigin(TreeNode.scala:82)
        at org.apache.spark.sql.catalyst.trees.TreeNode.transformDownWithPruning(TreeNode.scala:481)
        at org.apache.spark.sql.catalyst.plans.logical.LogicalPlan.org$apache$spark$sql$catalyst$plans$logical$AnalysisHelper$$super$transformDownWithPruning(LogicalPlan.scala:30)
        at org.apache.spark.sql.catalyst.plans.logical.AnalysisHelper.transformDownWithPruning(AnalysisHelper.scala:267)
        at org.apache.spark.sql.catalyst.plans.logical.AnalysisHelper.transformDownWithPruning$(AnalysisHelper.scala:263)
        at org.apache.spark.sql.catalyst.plans.logical.LogicalPlan.transformDownWithPruning(LogicalPlan.scala:30)
        at org.apache.spark.sql.catalyst.plans.logical.LogicalPlan.transformDownWithPruning(LogicalPlan.scala:30)
        at org.apache.spark.sql.catalyst.trees.TreeNode.transformDown(TreeNode.scala:457)
        at org.apache.spark.sql.execution.QueryExecution.eagerlyExecuteCommands(QueryExecution.scala:93)
        at org.apache.spark.sql.execution.QueryExecution.commandExecuted$lzycompute(QueryExecution.scala:80)
        at org.apache.spark.sql.execution.QueryExecution.commandExecuted(QueryExecution.scala:78)
        at org.apache.spark.sql.execution.QueryExecution.assertCommandExecuted(QueryExecution.scala:115)
        at org.apache.spark.sql.DataFrameWriter.runCommand(DataFrameWriter.scala:848)
        at org.apache.spark.sql.DataFrameWriter.saveToV1Source(DataFrameWriter.scala:382)
        at org.apache.spark.sql.DataFrameWriter.saveInternal(DataFrameWriter.scala:355)
        at org.apache.spark.sql.DataFrameWriter.save(DataFrameWriter.scala:239)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:566)
        at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:244)
        at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:357)
        at py4j.Gateway.invoke(Gateway.java:282)
        at py4j.commands.AbstractCommand.invokeMethod(AbstractCommand.java:132)
        at py4j.commands.CallCommand.execute(CallCommand.java:79)
        at py4j.ClientServerConnection.waitForCommands(ClientServerConnection.java:182)
        at py4j.ClientServerConnection.run(ClientServerConnection.java:106)
        at java.base/java.lang.Thread.run(Thread.java:829)
Caused by: java.lang.NullPointerException: Name is null
        at java.base/java.lang.Enum.valueOf(Enum.java:238)
        at org.apache.hudi.metrics.datadog.DatadogHttpClient$ApiSite.valueOf(DatadogHttpClient.java:120)
        at org.apache.hudi.config.HoodieWriteConfig.getDatadogApiSite(HoodieWriteConfig.java:2167)
        at org.apache.hudi.metrics.datadog.DatadogMetricsReporter.<init>(DatadogMetricsReporter.java:48)
        at org.apache.hudi.metrics.MetricsReporterFactory.createReporter(MetricsReporterFactory.java:78)
        at org.apache.hudi.metrics.Metrics.<init>(Metrics.java:58)
        at org.apache.hudi.metrics.Metrics.getInstance(Metrics.java:82)
        at org.apache.hudi.metrics.HoodieMetrics.<init>(HoodieMetrics.java:95)
        at org.apache.hudi.client.BaseHoodieClient.<init>(BaseHoodieClient.java:96)
        at org.apache.hudi.client.BaseHoodieWriteClient.<init>(BaseHoodieWriteClient.java:163)
        at org.apache.hudi.client.SparkRDDWriteClient.<init>(SparkRDDWriteClient.java:86)
        at org.apache.hudi.client.SparkRDDWriteClient.<init>(SparkRDDWriteClient.java:75)
        at org.apache.hudi.metadata.SparkHoodieBackedTableMetadataWriter.initializeWriteClient(SparkHoodieBackedTableMetadataWriter.java:146)
        at org.apache.hudi.metadata.HoodieBackedTableMetadataWriter.getWriteClient(HoodieBackedTableMetadataWriter.java:1474)
        at org.apache.hudi.metadata.HoodieBackedTableMetadataWriter.commitInternal(HoodieBackedTableMetadataWriter.java:1062)
        at org.apache.hudi.metadata.SparkHoodieBackedTableMetadataWriter.bulkCommit(SparkHoodieBackedTableMetadataWriter.java:130)
        at org.apache.hudi.metadata.HoodieBackedTableMetadataWriter.initializeFromFilesystem(HoodieBackedTableMetadataWriter.java:438)
        at org.apache.hudi.metadata.HoodieBackedTableMetadataWriter.initializeIfNeeded(HoodieBackedTableMetadataWriter.java:271)
        at org.apache.hudi.metadata.HoodieBackedTableMetadataWriter.<init>(HoodieBackedTableMetadataWriter.java:175)
        at org.apache.hudi.metadata.SparkHoodieBackedTableMetadataWriter.<init>(SparkHoodieBackedTableMetadataWriter.java:95)
        at org.apache.hudi.metadata.SparkHoodieBackedTableMetadataWriter.create(SparkHoodieBackedTableMetadataWriter.java:72)
        at org.apache.hudi.client.SparkRDDWriteClient.initializeMetadataTable(SparkRDDWriteClient.java:287)
        ... 50 more
```

--- a/hudi-client/hudi-client-common/src/main/java/org/apache/hudi/metadata/HoodieMetadataWriteUtils.java
+++ b/hudi-client/hudi-client-common/src/main/java/org/apache/hudi/metadata/HoodieMetadataWriteUtils.java
@@ -192,7 +192,8 @@ public class HoodieMetadataWriteUtils {
                   .withDatadogApiTimeoutSeconds(writeConfig.getDatadogApiTimeoutSeconds());
           if (writeConfig.getDatadogMetricHost() != null) {
             datadogConfig = datadogConfig.withDatadogHost(writeConfig.getDatadogMetricHost());
-          } else if (writeConfig.getDatadogApiSite() != null) {
+          }
+          if (writeConfig.getDatadogApiSite() != null) {
             datadogConfig = datadogConfig.withDatadogApiSite(writeConfig.getDatadogApiSite().name());
           }
 

--- a/hudi-client/hudi-client-common/src/main/java/org/apache/hudi/metadata/HoodieMetadataWriteUtils.java
+++ b/hudi-client/hudi-client-common/src/main/java/org/apache/hudi/metadata/HoodieMetadataWriteUtils.java
@@ -188,7 +188,7 @@ public class HoodieMetadataWriteUtils {
                   .withDatadogApiKeySkipValidation(writeConfig.getDatadogApiKeySkipValidation())
                   .withDatadogPrefix(writeConfig.getDatadogMetricPrefix())
                   .withDatadogReportPeriodSeconds(writeConfig.getDatadogReportPeriodSeconds())
-                  .withDatadogTags(String.join(";", writeConfig.getDatadogMetricTags()))
+                  .withDatadogTags(String.join(",", writeConfig.getDatadogMetricTags()))
                   .withDatadogApiTimeoutSeconds(writeConfig.getDatadogApiTimeoutSeconds());
           if (writeConfig.getDatadogMetricHost() != null) {
             datadogConfig = datadogConfig.withDatadogHost(writeConfig.getDatadogMetricHost());

--- a/hudi-client/hudi-client-common/src/main/java/org/apache/hudi/metadata/HoodieMetadataWriteUtils.java
+++ b/hudi-client/hudi-client-common/src/main/java/org/apache/hudi/metadata/HoodieMetadataWriteUtils.java
@@ -34,7 +34,11 @@ import org.apache.hudi.config.HoodieArchivalConfig;
 import org.apache.hudi.config.HoodieCleanConfig;
 import org.apache.hudi.config.HoodieCompactionConfig;
 import org.apache.hudi.config.HoodieWriteConfig;
-import org.apache.hudi.config.metrics.*;
+import org.apache.hudi.config.metrics.HoodieMetricsConfig;
+import org.apache.hudi.config.metrics.HoodieMetricsGraphiteConfig;
+import org.apache.hudi.config.metrics.HoodieMetricsJmxConfig;
+import org.apache.hudi.config.metrics.HoodieMetricsPrometheusConfig;
+import org.apache.hudi.config.metrics.HoodieMetricsDatadogConfig;
 import org.apache.hudi.exception.HoodieMetadataException;
 import org.apache.hudi.table.action.compact.strategy.UnBoundedCompactionStrategy;
 

--- a/hudi-client/hudi-client-common/src/main/java/org/apache/hudi/metadata/HoodieMetadataWriteUtils.java
+++ b/hudi-client/hudi-client-common/src/main/java/org/apache/hudi/metadata/HoodieMetadataWriteUtils.java
@@ -34,10 +34,7 @@ import org.apache.hudi.config.HoodieArchivalConfig;
 import org.apache.hudi.config.HoodieCleanConfig;
 import org.apache.hudi.config.HoodieCompactionConfig;
 import org.apache.hudi.config.HoodieWriteConfig;
-import org.apache.hudi.config.metrics.HoodieMetricsConfig;
-import org.apache.hudi.config.metrics.HoodieMetricsGraphiteConfig;
-import org.apache.hudi.config.metrics.HoodieMetricsJmxConfig;
-import org.apache.hudi.config.metrics.HoodieMetricsPrometheusConfig;
+import org.apache.hudi.config.metrics.*;
 import org.apache.hudi.exception.HoodieMetadataException;
 import org.apache.hudi.table.action.compact.strategy.UnBoundedCompactionStrategy;
 
@@ -182,6 +179,17 @@ public class HoodieMetadataWriteUtils {
           builder.withProperties(prometheusConfig.getProps());
           break;
         case DATADOG:
+          HoodieMetricsDatadogConfig datadogConfig = HoodieMetricsDatadogConfig.newBuilder()
+                  .withDatadogApiKey(writeConfig.getDatadogApiKey())
+                  .withDatadogApiKeySkipValidation(writeConfig.getDatadogApiKeySkipValidation())
+                  .withDatadogHost(writeConfig.getDatadogMetricHost())
+                  .withDatadogPrefix(writeConfig.getDatadogMetricPrefix())
+                  .withDatadogReportPeriodSeconds(writeConfig.getDatadogReportPeriodSeconds())
+                  .withDatadogTags(String.join(";", writeConfig.getDatadogMetricTags()))
+                  .withDatadogApiTimeoutSeconds(writeConfig.getDatadogApiTimeoutSeconds())
+                  .build();
+          builder.withProperties(datadogConfig.getProps());
+          break;
         case PROMETHEUS:
         case CONSOLE:
         case INMEMORY:

--- a/hudi-client/hudi-client-common/src/main/java/org/apache/hudi/metadata/HoodieMetadataWriteUtils.java
+++ b/hudi-client/hudi-client-common/src/main/java/org/apache/hudi/metadata/HoodieMetadataWriteUtils.java
@@ -179,16 +179,20 @@ public class HoodieMetadataWriteUtils {
           builder.withProperties(prometheusConfig.getProps());
           break;
         case DATADOG:
-          HoodieMetricsDatadogConfig datadogConfig = HoodieMetricsDatadogConfig.newBuilder()
+          HoodieMetricsDatadogConfig.Builder datadogConfig = HoodieMetricsDatadogConfig.newBuilder()
                   .withDatadogApiKey(writeConfig.getDatadogApiKey())
                   .withDatadogApiKeySkipValidation(writeConfig.getDatadogApiKeySkipValidation())
-                  .withDatadogHost(writeConfig.getDatadogMetricHost())
                   .withDatadogPrefix(writeConfig.getDatadogMetricPrefix())
                   .withDatadogReportPeriodSeconds(writeConfig.getDatadogReportPeriodSeconds())
                   .withDatadogTags(String.join(";", writeConfig.getDatadogMetricTags()))
-                  .withDatadogApiTimeoutSeconds(writeConfig.getDatadogApiTimeoutSeconds())
-                  .build();
-          builder.withProperties(datadogConfig.getProps());
+                  .withDatadogApiTimeoutSeconds(writeConfig.getDatadogApiTimeoutSeconds());
+          if (writeConfig.getDatadogMetricHost() != null) {
+            datadogConfig = datadogConfig.withDatadogHost(writeConfig.getDatadogMetricHost());
+          } else if (writeConfig.getDatadogApiSite() != null) {
+            datadogConfig = datadogConfig.withDatadogApiSite(writeConfig.getDatadogApiSite().name());
+          }
+
+          builder.withProperties(datadogConfig.build().getProps());
           break;
         case PROMETHEUS:
         case CONSOLE:

--- a/hudi-spark-datasource/hudi-spark/src/test/scala/org/apache/hudi/functional/TestMetricsReporter.scala
+++ b/hudi-spark-datasource/hudi-spark/src/test/scala/org/apache/hudi/functional/TestMetricsReporter.scala
@@ -1,0 +1,103 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hudi.functional
+
+import org.apache.hudi.HoodieConversionUtils.toJavaOption
+import org.apache.hudi.common.config.{HoodieMetadataConfig, HoodieStorageConfig}
+import org.apache.hudi.common.testutils.RawTripTestPayload.recordsToStrings
+import org.apache.hudi.common.util
+import org.apache.hudi.config.HoodieWriteConfig
+import org.apache.hudi.config.metrics.{HoodieMetricsConfig, HoodieMetricsDatadogConfig}
+import org.apache.hudi.testutils.HoodieSparkClientTestBase
+import org.apache.hudi.util.JFunction
+import org.apache.hudi.{DataSourceReadOptions, DataSourceWriteOptions, HoodieSparkRecordMerger, SparkDatasetMixin}
+import org.apache.spark.sql._
+import org.apache.spark.sql.hudi.HoodieSparkSessionExtension
+import org.junit.jupiter.api.{AfterEach, BeforeEach, Test}
+import org.slf4j.LoggerFactory
+
+import java.util.function.Consumer
+import scala.collection.JavaConverters._
+
+/**
+ * Tests on Spark DataSource for MOR table.
+ */
+class TestMetricsReporter extends HoodieSparkClientTestBase with SparkDatasetMixin {
+
+  var spark: SparkSession = null
+  private val log = LoggerFactory.getLogger(classOf[TestMORDataSource])
+  val commonOpts = Map(
+    "hoodie.insert.shuffle.parallelism" -> "4",
+    "hoodie.upsert.shuffle.parallelism" -> "4",
+    DataSourceWriteOptions.RECORDKEY_FIELD.key -> "_row_key",
+    DataSourceWriteOptions.PARTITIONPATH_FIELD.key -> "partition",
+    DataSourceWriteOptions.PRECOMBINE_FIELD.key -> "timestamp",
+    HoodieWriteConfig.TBL_NAME.key -> "hoodie_test"
+  )
+  val sparkOpts = Map(
+    HoodieWriteConfig.RECORD_MERGER_IMPLS.key -> classOf[HoodieSparkRecordMerger].getName,
+    HoodieStorageConfig.LOGFILE_DATA_BLOCK_FORMAT.key -> "parquet"
+  )
+
+  val verificationCol: String = "driver"
+  val updatedVerificationVal: String = "driver_update"
+
+  @BeforeEach override def setUp() {
+    setTableName("hoodie_test")
+    initPath()
+    initSparkContexts()
+    spark = sqlContext.sparkSession
+    initTestDataGenerator()
+    initFileSystem()
+  }
+
+  @AfterEach override def tearDown() = {
+    cleanupSparkContexts()
+    cleanupTestDataGenerator()
+    cleanupFileSystem()
+  }
+
+  override def getSparkSessionExtensionsInjector: util.Option[Consumer[SparkSessionExtensions]] =
+    toJavaOption(
+      Some(
+        JFunction.toJavaConsumer((receiver: SparkSessionExtensions) => new HoodieSparkSessionExtension().apply(receiver)))
+    )
+
+  @Test
+  def testSmokeDatadogReporter() {
+    val records1 = recordsToStrings(dataGen.generateInserts("001", 100)).asScala
+    val inputDF1 = spark.read.json(spark.sparkContext.parallelize(records1, 2))
+    val writeOpts: Map[String, String] = commonOpts ++ Map(
+      DataSourceReadOptions.ENABLE_HOODIE_FILE_INDEX.key -> "",
+      DataSourceWriteOptions.OPERATION.key -> DataSourceWriteOptions.UPSERT_OPERATION_OPT_VAL,
+      DataSourceWriteOptions.TABLE_TYPE.key -> DataSourceWriteOptions.COW_TABLE_TYPE_OPT_VAL,
+      HoodieMetadataConfig.ENABLE.key -> "true",
+      HoodieMetricsConfig.TURN_METRICS_ON.key -> "true",
+      HoodieMetricsConfig.METRICS_REPORTER_TYPE_VALUE.key -> "DATADOG",
+      HoodieMetricsDatadogConfig.API_KEY_SKIP_VALIDATION.key -> "true",
+      HoodieMetricsDatadogConfig.METRIC_PREFIX_VALUE.key -> "hudi",
+      HoodieMetricsDatadogConfig.API_SITE_VALUE.key -> "US",
+      HoodieMetricsDatadogConfig.API_KEY.key -> "dummykey",
+    )
+    inputDF1.write.format("org.apache.hudi")
+      .options(writeOpts)
+      .mode(SaveMode.Overwrite)
+      .save(basePath)
+
+  }
+}

--- a/hudi-spark-datasource/hudi-spark/src/test/scala/org/apache/hudi/functional/TestMetricsReporter.scala
+++ b/hudi-spark-datasource/hudi-spark/src/test/scala/org/apache/hudi/functional/TestMetricsReporter.scala
@@ -18,14 +18,14 @@
 package org.apache.hudi.functional
 
 import org.apache.hudi.HoodieConversionUtils.toJavaOption
-import org.apache.hudi.common.config.{HoodieMetadataConfig, HoodieStorageConfig}
+import org.apache.hudi.common.config.HoodieMetadataConfig
 import org.apache.hudi.common.testutils.RawTripTestPayload.recordsToStrings
 import org.apache.hudi.common.util
 import org.apache.hudi.config.HoodieWriteConfig
 import org.apache.hudi.config.metrics.{HoodieMetricsConfig, HoodieMetricsDatadogConfig}
 import org.apache.hudi.testutils.HoodieSparkClientTestBase
 import org.apache.hudi.util.JFunction
-import org.apache.hudi.{DataSourceReadOptions, DataSourceWriteOptions, HoodieSparkRecordMerger, SparkDatasetMixin}
+import org.apache.hudi.{DataSourceWriteOptions, SparkDatasetMixin}
 import org.apache.spark.sql._
 import org.apache.spark.sql.hudi.HoodieSparkSessionExtension
 import org.junit.jupiter.api.{AfterEach, BeforeEach, Test}
@@ -38,7 +38,6 @@ import scala.collection.JavaConverters._
  * Tests on Spark DataSource for MOR table.
  */
 class TestMetricsReporter extends HoodieSparkClientTestBase with SparkDatasetMixin {
-
   var spark: SparkSession = null
   private val log = LoggerFactory.getLogger(classOf[TestMORDataSource])
   val commonOpts = Map(
@@ -49,13 +48,6 @@ class TestMetricsReporter extends HoodieSparkClientTestBase with SparkDatasetMix
     DataSourceWriteOptions.PRECOMBINE_FIELD.key -> "timestamp",
     HoodieWriteConfig.TBL_NAME.key -> "hoodie_test"
   )
-  val sparkOpts = Map(
-    HoodieWriteConfig.RECORD_MERGER_IMPLS.key -> classOf[HoodieSparkRecordMerger].getName,
-    HoodieStorageConfig.LOGFILE_DATA_BLOCK_FORMAT.key -> "parquet"
-  )
-
-  val verificationCol: String = "driver"
-  val updatedVerificationVal: String = "driver_update"
 
   @BeforeEach override def setUp() {
     setTableName("hoodie_test")
@@ -83,7 +75,6 @@ class TestMetricsReporter extends HoodieSparkClientTestBase with SparkDatasetMix
     val records1 = recordsToStrings(dataGen.generateInserts("001", 100)).asScala
     val inputDF1 = spark.read.json(spark.sparkContext.parallelize(records1, 2))
     val writeOpts: Map[String, String] = commonOpts ++ Map(
-      DataSourceReadOptions.ENABLE_HOODIE_FILE_INDEX.key -> "",
       DataSourceWriteOptions.OPERATION.key -> DataSourceWriteOptions.UPSERT_OPERATION_OPT_VAL,
       DataSourceWriteOptions.TABLE_TYPE.key -> DataSourceWriteOptions.COW_TABLE_TYPE_OPT_VAL,
       HoodieMetadataConfig.ENABLE.key -> "true",
@@ -92,12 +83,11 @@ class TestMetricsReporter extends HoodieSparkClientTestBase with SparkDatasetMix
       HoodieMetricsDatadogConfig.API_KEY_SKIP_VALIDATION.key -> "true",
       HoodieMetricsDatadogConfig.METRIC_PREFIX_VALUE.key -> "hudi",
       HoodieMetricsDatadogConfig.API_SITE_VALUE.key -> "US",
-      HoodieMetricsDatadogConfig.API_KEY.key -> "dummykey",
-    )
+      HoodieMetricsDatadogConfig.API_KEY.key -> "dummykey")
+
     inputDF1.write.format("org.apache.hudi")
       .options(writeOpts)
       .mode(SaveMode.Overwrite)
       .save(basePath)
-
   }
 }

--- a/hudi-spark-datasource/hudi-spark/src/test/scala/org/apache/hudi/functional/TestMetricsReporter.scala
+++ b/hudi-spark-datasource/hudi-spark/src/test/scala/org/apache/hudi/functional/TestMetricsReporter.scala
@@ -28,7 +28,8 @@ import org.apache.hudi.util.JFunction
 import org.apache.hudi.{DataSourceWriteOptions, SparkDatasetMixin}
 import org.apache.spark.sql._
 import org.apache.spark.sql.hudi.HoodieSparkSessionExtension
-import org.junit.jupiter.api.{AfterEach, BeforeEach, Test}
+import org.junit.jupiter.api.function.Executable
+import org.junit.jupiter.api.{AfterEach, Assertions, BeforeEach, Test}
 import org.slf4j.LoggerFactory
 
 import java.util.function.Consumer
@@ -85,9 +86,13 @@ class TestMetricsReporter extends HoodieSparkClientTestBase with SparkDatasetMix
       HoodieMetricsDatadogConfig.API_SITE_VALUE.key -> "US",
       HoodieMetricsDatadogConfig.API_KEY.key -> "dummykey")
 
-    inputDF1.write.format("org.apache.hudi")
-      .options(writeOpts)
-      .mode(SaveMode.Overwrite)
-      .save(basePath)
+    Assertions.assertDoesNotThrow(new Executable {
+      override def execute(): Unit =
+        inputDF1.write.format("org.apache.hudi")
+          .options(writeOpts)
+          .mode(SaveMode.Overwrite)
+          .save(basePath)
+
+    })
   }
 }

